lib/nerves_network/dhclientv4.ex

```diff
@@ -0,0 +1,212 @@
+    Logger.debug "dhclient: deconfigure #{ifname}"
+
+    Utils.notify(Nerves.Dhclientv4, state.ifname, :deconfig, %{ifname: ifname})
+    {:noreply, state}
+  end
+
+  #Handling informational debug prints from the dhclient
+  defp handle_dhclient([message], state) do
+    Logger.debug fn -> "#{__MODULE__} handle_dhclient args = #{inspect message} state = #{inspect state}" end
+    {:noreply, state}
+  end
+
+  #TODO: scribe PREINIT6 handler
+  #[".../dhclient_wrapper", "PREINIT6", "eth1", "", "", ""] state = %{ifname: "eth1", port: #Port<0.5567>}
+
+  defp handle_dhclient([_originator, "REBIND", ifname, ip, domain_search, dns, old_ip], state) do
+    dnslist = String.split(dns, " ")
+    Logger.debug fn -> "dhclient: rebind #{ifname}: IPv4 = #{ip}, domain_search=#{domain_search}, dns=#{inspect dns} old_ip=#{inspect old_ip}" end
+    Utils.notify(Nerves.Dhclientv4, state.ifname, :rebind , %{ifname: ifname, ipv4_address: ip, ipv4_domain: domain_search, ipv4_nameservers: dnslist, old_ipv4_address: old_ip})
+    {:noreply, state}
+  end
+  defp handle_dhclient([_originator, "BOUND", ifname, ip, domain_search, dns, old_ip], state) do
+    dnslist = String.split(dns, " ")
+    Logger.debug fn -> "dhclient: bound #{ifname}: ipv4 = #{ip}, domain_search=#{domain_search}, dns=#{inspect dns}" end
+    Utils.notify(Nerves.Dhclientv4, state.ifname, :bound, %{ifname: ifname, ipv4_address: ip, ipv4_domain: domain_search, ipv4_nameservers: dnslist, old_ipv4_address: old_ip})
+    {:noreply, state}
+  end
+  defp handle_dhclient([_originator, "RENEW", ifname, ip, domain_search, dns, old_ip], state) do
+    dnslist = String.split(dns, " ")
+    Logger.debug "dhclient: renew #{ifname}"
+    Utils.notify(Nerves.Dhclientv4, state.ifname, :renew, %{ifname: ifname, ipv4_address: ip, ipv4_domain: domain_search, ipv4_nameservers: dnslist, old_ipv4_address: old_ip})
+    {:noreply, state}
+  end
+
+  defp handle_dhclient([_originator, "RELEASE", ifname, _ip, _domain_search, _dns, old_ip], state) do
+    Logger.debug fn -> "dhclient: release #{ifname}" end
+    Utils.notify(Nerves.Dhclientv4, state.ifname, :renew, %{ifname: ifname, old_ipv4_address: old_ip})
+    {:noreply, state}
+  end
+  defp handle_dhclient([_originator, "EXPIRE", ifname, _ip, _domain_search, _dns, old_ip], state) do
+    Logger.debug fn -> "dhclient: expire #{ifname}" end
+    Utils.notify(Nerves.Dhclientv4, state.ifname, :expire, %{ifname: ifname, old_ipv4_address: old_ip})
+    {:noreply, state}
+  end
+  defp handle_dhclient([_originator, "STOP", ifname, _ip, _domain_search, _dns, old_ip], state) do
+    Logger.debug fn -> "dhclient: stop #{ifname}" end
+    Utils.notify(Nerves.Dhclientv4, state.ifname, :stop, %{ifname: ifname, old_ipv4_address: old_ip})
+    {:noreply, state}
+  end
+
+  defp handle_dhclient(_something_else, state) do
+    #msg = List.foldl(something_else, "", &<>/2)
+    #Logger.debug "dhclient: ignoring unhandled message: #{msg}"
```

Comment:
We actually want to uncomment that Logger.debug but you need #{inspect something_else}